experimental/skills/DialogToDialog/DialogRootBot/Dialogs/MainDialog.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Integration.AspNet.Core.Skills;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+
+namespace Microsoft.BotBuilderSamples.DialogRootBot.Dialogs
+{
+    /// <summary>
+    /// A sample MainDialog that uses SkillDialog to call skills.
+    /// </summary>
+    public class MainDialog : ComponentDialog
+    {
+        // Note: in this example we using only one skill.
+        private const string _targetSkillId = "DialogSkillBot";
+        private readonly string _botId;
+        private readonly ConversationState _conversationState;
+        private readonly SkillHttpClient _skillClient;
+        private readonly SkillsConfiguration _skillsConfig;
+
+        // Dependency injection uses this constructor to instantiate MainDialog
+        public MainDialog(ConversationState conversationState, SkillHttpClient skillClient, SkillsConfiguration skillsConfig, SkillDialog skillDialog, IConfiguration configuration)
+            : base(nameof(MainDialog))
+        {
+            _botId = configuration.GetSection(MicrosoftAppCredentials.MicrosoftAppIdKey)?.Value;
+            _skillClient = skillClient;
+            _skillsConfig = skillsConfig;
+            _conversationState = conversationState ?? throw new ArgumentNullException(nameof(conversationState));
+            AddDialog(new TextPrompt(nameof(TextPrompt)));
+            AddDialog(skillDialog);
+            AddDialog(new WaterfallDialog(nameof(WaterfallDialog), new WaterfallStep[]
+            {
+                IntroStepAsync,
+                ActStepAsync,
+                FinalStepAsync
+            }));
+
+            // The initial child Dialog to run.
+            InitialDialogId = nameof(WaterfallDialog);
+        }
+
+        private async Task<DialogTurnResult> IntroStepAsync(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        {
+            // Use the text provided in FinalStepAsync or the default if it is the first time.
```

Comment:
```suggestion
            // Use the text provided in the dialog options if it is present; otherwise, use the default text.
```

Reply:
Updated comment

---
In reply to: [356221881](https://github.com/microsoft/BotBuilder-Samples/pull/2032#discussion_r356221881) [](ancestors = 356221881)